plan/logical_plan_builder.go

```diff
@@ -1332,6 +1380,21 @@ func (b *planBuilder) buildUpdateLists(tableList []*ast.TableName, list []*ast.A
 	return newList, p
 }
 
+// extractTableAsNameForUpdate extracts tables' name alias for update.
+func extractTableAsNameForUpdate(p Plan) (asNames map[*model.TableInfo]model.CIStr) {
```

Comment:
For SQL `update a from b a, (select * from a c) b`, will the tableInfo a match with name c ?